### ISSUE TITLE
Effective Sample Size useless computation

### ISSUE THREAD

**Issue by emilemathieu** [yebai/Turing.jl#350]
Hi all !

Since the function `weights` is defined as
```julia
function weights(pc :: ParticleContainer)
  @assert pc.num_particles == length(pc)
  logWs = pc.logWs
  Ws = exp(logWs-maximum(logWs))
  logZ = log(sum(Ws)) + maximum(logWs)
  Ws = Ws ./ sum(Ws)
  return Ws, logZ
end
```
why computing `sum(Ws) ^ 2` in `effectiveSampleSize` (reminded below) since its equals `1` ?
```julia
function effectiveSampleSize(pc :: ParticleContainer)
  Ws, _ = weights(pc)
  ess = sum(Ws) ^ 2 / sum(Ws .^ 2)
end
```

Best,
Emile

**Comment by yebai** [comment]
Nice catch - this can be safely removed. I guess `effectiveSampleSize` function was written first, later we introduced the `weights` function which also performs normalization. 

Could you correct a PR to remove the redundant `sum(Ws) ^ 2` in `effectiveSampleSize`?

**Comment by emilemathieu** [comment]
Sure, I'll do it tomorrow.

**Comment by emilemathieu** [comment]
See PR https://github.com/yebai/Turing.jl/pull/355

Best,
Emile

**Comment by yebai** [comment]
Fixed by #355.
